plugin/pkg/shoot/validator/admission.go

```diff
@@ -628,6 +634,107 @@ func (c *validationContext) validateShootHibernation(a admission.Attributes) err
 	return nil
 }
 
+func (c *validationContext) validateSecretBindingToCredentialsBindingMigration(
+	a admission.Attributes,
+	secretBindingLister gardencorev1beta1listers.SecretBindingLister,
+	credentialsBindingLister securityv1alpha1listers.CredentialsBindingLister,
+) error {
+	secretBindingNameProgressedToEmpty := c.oldShoot.Spec.SecretBindingName != nil && c.shoot.Spec.SecretBindingName == nil
+	credentialsBindingNameProgressedToSet := c.oldShoot.Spec.CredentialsBindingName == nil && c.shoot.Spec.CredentialsBindingName != nil
+
+	if secretBindingNameProgressedToEmpty && credentialsBindingNameProgressedToSet {
+		secretBinding, err := secretBindingLister.SecretBindings(c.oldShoot.Namespace).Get(*c.oldShoot.Spec.SecretBindingName)
+		if err != nil {
+			return apierrors.NewInternalError(fmt.Errorf("could not retrieve previously referenced secret binding: %+v", err.Error()))
```

Comment:
You can wrap the error here
```suggestion
			return apierrors.NewInternalError(fmt.Errorf("could not retrieve previously referenced secret binding: %w", err))
```

Reply:
Yes, but I chose this way so it is consistent with other returned errors from this validator.